src/main/webapp/app/components/curationPanel/CompanionDiagnosticDevicePanel.tsx

```diff
@@ -0,0 +1,123 @@
+import React, { useState } from 'react';
+import { Menu } from 'react-pro-sidebar';
+import { Button, Form } from 'reactstrap';
+import { SearchOptionType } from 'app/config/constants';
+import { useHistory, useLocation } from 'react-router-dom';
+import { notifyError, notifySuccess } from 'app/oncokb-commons/components/util/NotificationUtils';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faPlus } from '@fortawesome/free-solid-svg-icons/faPlus';
+import DefaultTooltip from 'app/shared/tooltip/DefaultTooltip';
+import CancerTypeSelect from 'app/shared/select/CancerTypeSelect';
+import { biomarkerAssociationClient } from 'app/shared/api/clients';
+import { SaveButton } from 'app/shared/button/SaveButton';
+import GeneSelect from 'app/shared/select/GeneSelect';
+import AlterationSelect from 'app/shared/select/AlterationSelect';
+import DrugSelect from 'app/shared/select/DrugSelect';
+import { BiomarkerAssociationDTO } from 'app/shared/api/generated';
+import FdaSubmissionSelect from 'app/shared/select/FdaSubmissionSelect';
+import { connect } from 'app/shared/util/typed-inject';
+import { IRootStore } from 'app/stores';
+
+const SidebarMenuItem: React.FunctionComponent<{ style?: React.CSSProperties }> = ({ style, children }) => {
+  return <div style={{ padding: '8px 24px 0 24px', ...style }}>{children}</div>;
+};
+
+export const defaultAdditional = {
+  page: 1,
+  type: SearchOptionType.GENE,
+};
+
+const CompanionDiagnosticDevicePanel: React.FunctionComponent<StoreProps> = props => {
+  const [selectedGeneId, setSelectedGeneId] = useState(null);
+  const [alterationValue, onAlterationChange] = useState(null);
+  const [cancerTypeValue, onCancerTypeChange] = useState(null);
+  const [drugValue, onDrugChange] = useState(null);
+  const [fdaSubmissionValue, onFdaSubmissionChange] = useState(null);
+
+  const history = useHistory();
+  const location = useLocation();
+  const id = parseInt(location.pathname.split('/')[2], 10);
+
+  const createBiomarkerAssociation = (e: any) => {
+    e.preventDefault();
+    const biomarkerAssociationDTO: BiomarkerAssociationDTO = {
+      fdaSubmissions: fdaSubmissionValue.map(fdaSubmission => fdaSubmission.value),
+      alterations: alterationValue.map(alteration => alteration.value),
+      cancerType: cancerTypeValue.value as number,
+      drugs: drugValue.map(drug => drug.value),
+      gene: selectedGeneId,
+    };
+    biomarkerAssociationClient
+      .createBiomarkerAssociation(biomarkerAssociationDTO)
+      .then(() => {
+        notifySuccess('Biomarker association added.');
+        props.getBiomarkerAssociations(id);
+      })
+      .catch(error => notifyError(error));
+  };
+
+  const redirectToCreateAlteration = () => {
+    history.push('/alteration/new');
```

Comment:
Do we have ROUTE constant that manages all these strings?

Reply:
Updated to use the util function that generates the route based on the entity type and action.